Opendkim development

stephdl opened the pull request:
backend is done

WIP for the frontend

NethServer/dev#5407

stephdl committed:
backed side for opendkim

stephdl committed:
create a key of 2048 bit

stephdl committed:
find smtpd_milters in service property by smtpdMilter

stephdl committed:
first UI, no Trusted hosts

stephdl committed:
chown opendkim:adm and chmod 640 *.txt

stephdl committed:
translation

stephdl committed:
new UI with text area for IP/CIDR List

stephdl committed:
action corrected, thank filippo

stephdl committed:
cleaned comment

stephdl committed:
report fails and default key 1024

stephdl committed:
New UI with a button to configure opendkim

stephdl committed:
changed event to update

stephdl committed:
added a different error message

stephdl committed:
modified TrustedHosts due UI modification

stephdl committed:
createlink typo

stephdl committed:
translation minor

stephdl committed:
removed the array in module

stephdl committed:
removed \n TrustedHosts

stephdl committed:
moved keys backup to backup-config.d

stephdl committed:
moved smtpdMilter  to SmtpdMilter property

stephdl committed:
moved smtpdMilter  to SmtpdMilter property

stephdl committed:
moved openDkim  to OpenDkim property

stephdl committed:
moved openDkim  to OpenDkim property, translation modifications

DavidePrincipi commented on root/etc/e-smith/templates/etc/postfix/main.cf/70SmtpdMilters:
```diff
@@ -0,0 +1,37 @@
+my @services = map { $_->key } $confDb->get_all_by_prop('type' => 'service');
+my @milters;
+
+foreach (@services) {
```
Initialize the `@milters` array in earlier template fragments added by additional RPMs

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Mail_Domain.php:
```diff
@@ -12,3 +12,6 @@
 $L['valid_relay_notprimarydomain'] = 'The local mailbox domain cannot be relayed';
+$L['OpenDkim_label'] = 'DomainKeys Identified Mail (DKIM)';
+$L['DkimKey_label'] = 'Add a TXT record to your external dns provider with the  selector default._domainkey and the key ("v=DKIM1; k=rsa; p=..;)"';
+$L['OpenDkimService_label'] = 'Dkim settings';
```
Always write DKIM in upper-case

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Mail_Domain_OpenDkimService.php:
```diff
@@ -0,0 +1,8 @@
+$L['openDkimRestrictedIpList_label'] = 'List of IP/CIDR (local network are allowed)';
+$L['Not an IP'] = '${0} is not an IP or a network CIDR (192.168.1.0/24)';
+$L['status_label'] = 'Enable the service';
+$L['Opendkim_Configure_header'] = 'Opendkim settings';
```
Always write DKIM in upper-case

DavidePrincipi commented on root/etc/e-smith/db/configuration/defaults/opendkim/SmtpdMilter:
```diff
@@ -0,0 +1 @@
+inet:127.0.0.1:8891
```
Should not be needed if `@milters` is initialized with it in the template code

DavidePrincipi reviewed (changes requested):
A reminder for the `@milters` initialization refactor plus minor remarks about using DKIM in uppercase in the UI labels. 

I still have some doubts around the "use DKIM from every/certain hosts with authenticated SMTP users" radio button 🤔

stephdl committed:
thank davidep, new way to write the smtpd_milters

stephdl committed:
added milter_mail_macros by pushing an array

stephdl committed:
dkim only sign, verification removed

stephdl committed:
all hosts are  enabled signing  per default in TrustedHosts

stephdl committed:
keys created without prop OpenDkim condition

stephdl committed:
Translation changed for OpenDkim checkbox

stephdl committed:
add different view following domain-create or domain-modify

stephdl committed:
create dkim key for previously created domain

stephdl committed:
removal of the dkim button in domain table

DavidePrincipi commented on root/etc/e-smith/templates/etc/opendkim/TrustedHosts/10base:
```diff
@@ -0,0 +1,26 @@
+# The localhost IP (127.0.0.1) should always be the first entry in this file.
+127.0.0.1
+::1
+{
```
A static `0.0.0.0/0` catches everything: I'd remove this block

stephdl replied:
done, but localhost must be there as asked by comment

DavidePrincipi commented on root/etc/e-smith/db/configuration/defaults/postfix/milterAction:
```diff
@@ -0,0 +1 @@
+accept
```
Upper case as other props? i.e. `MilterAction`
 
I think this prop could be removed, for the same reason of `SmtpdMilter`

stephdl replied:
this prop has been removed, the action is forced to 'reject', and the smtpd_milter failure reason must be found.

DavidePrincipi commented on root/etc/e-smith/events/actions/nethserver-mail-common-openDkim-Key-Creation:
```diff
@@ -0,0 +1,23 @@
+}
+
+#chmod and chown to allow the server-manager to read the txt file
+system ('/usr/bin/chmod 640 /etc/opendkim/keys/*.txt') == '0'||
```
Does the action/event fail if the glob expression does not match anything?

stephdl replied:
yes the action fails if we remove the keys, or if we cannot create the keys

Jan 10 22:14:27 ns7dev4 esmith::event[16439]: /usr/bin/chmod: cannot access ‘/etc/opendkim/keys/*.txt’: No such file or directory
Jan 10 22:14:27 ns7dev4 esmith::event[16439]: the /etc/opendkim/keys/*.txt cannot be chmod 640 : 256 at /etc/e-smith/events/nethserver-mail-common-update/S40nethserver-mail-common-openDkim-Key-Creation line 19.
Jan 10 22:14:27 ns7dev4 esmith::event[16439]: Action: /etc/e-smith/events/nethserver-mail-common-update/S40nethserver-mail-common-openDkim-Key-Creation FAILED: 1 [0.175246]



Jan 10 22:12:20 ns7dev4 esmith::event[16240]: opendkim-genkey: /etc/opendkim/keys/: chdir(): No such file or directory at /usr/sbin/opendkim-genkey line 125.
Jan 10 22:12:20 ns7dev4 esmith::event[16240]: Opendkim cannot create keys for nethservertest.org: 512 at /etc/e-smith/events/nethserver-mail-common-update/S40nethserver-mail-common-openDkim-Key-Creation line 14.
Jan 10 22:12:20 ns7dev4 esmith::event[16240]: Action: /etc/e-smith/events/nethserver-mail-common-update/S40nethserver-mail-common-openDkim-Key-Creation FAILED: 2 [0.172427]

DavidePrincipi commented on root/etc/e-smith/templates/etc/opendkim.conf/10base:
```diff
@@ -0,0 +1,135 @@
+##  Specifies whether or not the filter should generate report mail back
+##  to senders when verification fails and an address for such a purpose
+##  is provided. See opendkim.conf(5) for details.
+SendReports	yes
```
Even if verification is disabled I wouldn't report back anything to senders

stephdl replied:
done

DavidePrincipi commented on root/etc/e-smith/templates/etc/opendkim.conf/10base:
```diff
@@ -0,0 +1,135 @@
+##  policies retrieved from DNS, rather than relying on the nameserver for
+##  caching service. Useful if the nameserver being used by the filter is
+##  not local.
+QueryCache	yes
```
We actually have a local DNS cache (dnsmasq's `cache-size=4000`), can we disable it here?

stephdl replied:
yes mostly because it is used for verification....so definitively yes

DavidePrincipi commented on root/etc/e-smith/templates/etc/opendkim/KeyTable/10base:
```diff
@@ -0,0 +1,17 @@
+
+    foreach my $domain (@domains) {
+        $domainName = $domain->key;
+        $dkim =  $domain->prop('OpenDkim') || 'disabled';
```
Informative note (no need to change IMO): in the past I used the convention to add a `*Status` suffix to props accepting value `enabled` or `disabled`, i.e. `OpenDkimStatus`

stephdl replied:
better to get the same convention, renamed to OpenDkimStatus

DavidePrincipi commented on root/etc/e-smith/templates/etc/postfix/main.cf/70smtpd_milters:
```diff
@@ -0,0 +1,13 @@
+# default action: accept (skip milters if something goes wrong)
+# else set to 'reject'
+
+milter_default_action = {$postfix{milterAction} || 'accept'}
```
I'd remove the comment block, this line and the prop, to retain the default Postfix value `tempfail`. If a milter fails there's an issue that must be fixed before going on.

stephdl replied:
done but I don't know how postfix warns the sysadmin , do you have some clues ?

DavidePrincipi commented on root/etc/e-smith/templates/etc/postfix/main.cf/01smtp_milters_opendkim:
```diff
@@ -0,0 +1,7 @@
+
+'';
+}
+
```
The trailing blanks are expanded! Should be trimmed.

I'd move this fragment code inside the `00template_vars` though, to keep the fragments more compact.

...and also set the UNIX socket path!

stephdl replied:
done, but for now the socket tcp is still there

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Mail_Domain.php:
```diff
@@ -12,3 +12,7 @@
 $L['domain_label'] = 'Domain';
 $L['valid_relay_notprimarydomain'] = 'The local mailbox domain cannot be relayed';
+$L['OpenDkim_label'] = 'Sign outbound messages with DomainKeys Identified Mail (DKIM)';
+$L['DkimKey_label'] = 'Add a TXT record to your external dns provider with the  selector default._domainkey and the key ("v=DKIM1; k=rsa; p=..;)"';
```
- `dns` should be `DNS`
- the ending `"` char does not match with parentheses

stephdl replied:
done

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Mail_Domain.php:
```diff
@@ -12,3 +12,7 @@
 $L['valid_relay_notprimarydomain'] = 'The local mailbox domain cannot be relayed';
+$L['OpenDkim_label'] = 'Sign outbound messages with DomainKeys Identified Mail (DKIM)';
+$L['DkimKey_label'] = 'Add a TXT record to your external dns provider with the  selector default._domainkey and the key ("v=DKIM1; k=rsa; p=..;)"';
+$L['OpenDkimService_label'] = 'Dkim settings';
```
Is it still used somewhere?

stephdl replied:
done

DavidePrincipi commented on createlinks:
```diff
@@ -89,13 +100,24 @@ event_actions('migration-import', qw(
+foreach (qw(create modify delete)) {
+    event_templates('host-' . $_, '/etc/opendkim/TrustedHosts');
+    event_services('host-'  . $_, 'opendkim' => 'restart');
+}
```
We can remove this block if TrustedHosts is static

DavidePrincipi commented on createlinks:
```diff
@@ -89,13 +100,24 @@ event_actions('migration-import', qw(
 #
 event_templates('trusted-networks-modify', qw(
     /etc/postfix/mynetworks.cidr
+    /etc/opendkim/TrustedHosts
```
We can remove this template expansion if TrustedHosts is static

DavidePrincipi commented on createlinks:
```diff
@@ -89,13 +100,24 @@ event_actions('migration-import', qw(
 event_services('trusted-networks-modify', qw(
     postfix reload
     amavisd reload
+    opendkim restart
```
We can remove this service restart if TrustedHosts is static

DavidePrincipi commented on root/etc/e-smith/db/configuration/defaults/opendkim/TCPPort:
```diff
@@ -0,0 +1 @@
+8891
```
If the service is provided only to localhost we can remove it: the firewall configuration always allows connections from localhost to localhost.

I'd configure a UNIX socket, which provides filesystem-level access control.

DavidePrincipi commented on root/etc/e-smith/templates/etc/opendkim.conf/10base:
```diff
@@ -0,0 +1,135 @@
+UserID	opendkim:opendkim
+
+##  Create a socket through which your MTA can communicate.
+Socket	inet:8891@localhost
```
I'd configure a UNIX socket, which provides filesystem-level access control. For instance `/var/run/opendkim.sock`

stephdl replied:
like said above, it is workable if I add postfix to the opendkim group, or worse, with umask 000 :'(

DavidePrincipi commented on root/etc/e-smith/templates/etc/opendkim.conf/10base:
```diff
@@ -0,0 +1,135 @@
+LogWhy	yes
+
+##  Attempt to become the specified user before starting operations.
+UserID	opendkim:opendkim
```
We can set it to `opendkim:mail` to grant Postfix daemon access to UNIX socket. Indeed Postfix is member of group `mail` too.

stephdl replied:
well it fails the permission are still 
-rw-r--r-- 1 opendkim opendkim 6 Jan 10 21:46 opendkim.pid
srwxrwxr-x 1 opendkim opendkim 0 Jan 10 21:46 opendkim.sock

if we add postfix to the opendkim group it is workable... :?

DavidePrincipi replied:
`UserID` works if the process starts with root privileges. But the systemd unit starts it with user privileges. We need to adjust some permissions... Let me think of it

DavidePrincipi replied:
We need to fix some permissions to make it work

```text
[root@vm5 ~]# rpm -V opendkim
S.5....T.  c /etc/opendkim.conf
SM5..UGT.  c /etc/opendkim/KeyTable
SM5..UGT.  c /etc/opendkim/SigningTable
SM5..UGT.  c /etc/opendkim/TrustedHosts
......G..    /etc/opendkim/keys
.M.......    /var/run/opendkim
```

- Templates can be fixed by adding templates.metadata configuration properly
- The temporary dir `/var/run/opendkim` is a wrong configuration of systemd tmpfiles.d feature. We can report it to the RPM maintainer, and (temporarly) fix it as we like with an override file.
  > Packages should install their configuration files in /usr/lib/tmpfiles.d. Files in /etc/tmpfiles.d are reserved for the local administrator (tmpfiles.d manpage)
- Add a systemd unit override fragment to start with `mail` group: in `/etc/systemd/system/opendkim.service.d/nethserver.conf`
  ```text
  [Service]
  Group=mail
  ```
- Set `Socket` to `/var/run/opendkim/milter`
- Umask can be restricted to 007

DavidePrincipi replied:
To fix the temporary dir permissions I added an override fragment for tmpfiles.d config:

```
[root@vm5 ~]# cat /etc/tmpfiles.d/opendkim-00nethserver.conf 
D /var/run/opendkim 0775 opendkim opendkim -
```

It should survive to reboots (/var/run is a tmpfs and dirs are created by systemd-tmpfiles, see manpage)

DavidePrincipi commented on root/etc/e-smith/templates/etc/opendkim.conf/10base:
```diff
@@ -0,0 +1,135 @@
+##  Specifies the sending address to be used on From: headers of outgoing
+##  failure reports.  By default, the e-mail address of the user executing
+##  the filter is used (executing_user@hostname).
+ReportAddress	"{$DomainName} Postmaster" <postmaster@{$DomainName}>
```
Comment out this one too

stephdl replied:
done

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php:
```diff
@@ -64,6 +71,21 @@ public function validate(\Nethgui\Controller\ValidationReportInterface $report)
+            return '';
+        }
+        $fileName =  self::DKIM_PATH . $this->parameters['domain'] . '.txt';
+        $value = $this->getPhpWrapper()->file_get_contents($fileName, FALSE, NULL, -1, self::DKIM_MAX_LENGTH);
```
Do not limit the read size: let's read the whole file contents!

stephdl replied:
done

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php:
```diff
@@ -35,6 +35,8 @@ class Modify extends \Nethgui\Controller\Table\Modify
     const DISCLAIMER_MAX_LENGTH = 2048;
     const DISCLAIMER_PATH = '/var/lib/nethserver/mail-disclaimers/';
+    const DKIM_PATH = '/etc/opendkim/keys/';
+    const DKIM_MAX_LENGTH = 2048;
```
Shouldn't be necessary

stephdl replied:
done

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php:
```diff
@@ -43,15 +45,20 @@ public function initialize()
                 array($this, 'readDisclaimerFile'), array($this, 'writeDisclaimerFile'), array()
             ));
 
+        $this->declareParameter('DkimKey', $this->createValidator()->maxLength(self::DKIM_MAX_LENGTH),$this->getPlatform()->getMapAdapter(
```
Let's set it to a large enough value, say 10000.

stephdl replied:
done

DavidePrincipi reviewed (changes requested):
I'd like to configure a UNIX socket, because OpenDKIM seems to support it.

Also other cleanups are suggested.

stephdl committed:
translations typo and changes

stephdl committed:
no limitation for reading and increase maximum parameter length

stephdl committed:
reject email when smtpd_milter fails

stephdl committed:
smtpd_milters array is in 00template_vars

stephdl committed:
0.0.0.0/0 catches everything, hostnames are removed

stephdl committed:
domain dkim prop renamed to OpenDkimStatus

stephdl committed:
do not query cache, do not send report

stephdl committed:
TrustedHosts is static, no templates expand or service restart

stephdl committed:
human key reading enhancement

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php:
```diff
@@ -64,6 +69,21 @@ public function validate(\Nethgui\Controller\ValidationReportInterface $report)
+            return '';
+        }
+        $fileName =  '/etc/opendkim/keys/' . $this->parameters['domain'] . '.txt';
+        $value = $this->getPhpWrapper()->file_get_contents($fileName, FALSE, NULL, -1);
```
If we fix the dir permissions I'm afraid we do not have the rights to read that file directly and we need to implement a sudo helper script :(

stephdl committed:
smtp_milter back

stephdl committed:
typo in templates and translation language

stephdl committed:
template.metadata created for /etc/opendkim/

stephdl committed:
switch to unix socket for opendkim

stephdl committed:
TCPPort prop of opendkim is removed

stephdl committed:
no more chown/chmod, key.txt place is readable to all

stephdl committed:
handle the case where keys are not created

stephdl committed:
renamed key creation action

stephdl committed:
opendkim uses one unique key for all domain

stephdl committed:
unique key but we must control when to sign

stephdl committed:
unique key created only with the -update event

stephdl committed:
chown, chmod, cp public key all the time

stephdl committed:
Display the unique key with domain-create

stephdl committed:
perl way to chown and chmod keys

stephdl committed:
New exeption or instead of ||

stephdl committed:
copyright added to action

stephdl committed:
removed the glob, use a static file name

DavidePrincipi commented on root/etc/e-smith/events/actions/nethserver-mail-common-opendkim-key-creation:
```diff
@@ -0,0 +1,51 @@
+
+my $db = esmith::ConfigDB->open_ro() or die("Could not open ConfigDB");
+my $domainName = $db->get_value('DomainName') ||
+    die "Opendkim cannot determine host's domain name: $?";
```
Failing actions should prefix log messages with `[ERROR]`

Other common prefixes are `[WARNING]` and `[NOTICE]`

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Mail_Domain.php:
```diff
@@ -12,3 +12,8 @@
+$L['OpenDkimStatus_label'] = 'Sign outbound messages with DomainKeys Identified Mail (DKIM)';
+$L['DkimKey_label'] = 'Copy and paste the following text in the record data (RDATA) section:';
+$L['DkimDNSTips_label'] = 'Add a TXT record to your public DNS service provider:';
+$L['DkimSelector_label'] = 'Set the record name to';
```
Can we merge the two labels into one (see patch below)

```
   Add a TXT record to your public DNS service provider with key "${0}"
```

DavidePrincipi commented on root/usr/share/nethesis/NethServer/Template/Mail/Domain/Modify.php:
```diff
@@ -42,6 +42,11 @@
+echo $view->fieldsetSwitch('OpenDkimStatus', 'enabled', $view::FIELDSETSWITCH_EXPANDABLE | $view::FIELDSETSWITCH_CHECKBOX)
+    ->setAttribute('uncheckedValue', 'disabled')
+    ->insert($view->literal($T('DkimDNSTips_label').'<br /><br />'))
+    ->insert($view->literal($T('DkimSelector_label').' : "default._domainkey"<br /><br />'))
```
```patch
From c2537bac60c6868b7744daedc60300225c777564 Mon Sep 17 00:00:00 2001
From: Davide Principi <davide.principi@nethesis.it>
Date: Mon, 15 Jan 2018 14:48:45 +0100
Subject: [PATCH] Compact UI label

---
 .../NethServer/Language/en/NethServer_Module_Mail_Domain.php        | 6 ++----
 root/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php    | 1 +
 root/usr/share/nethesis/NethServer/Template/Mail/Domain/Modify.php  | 3 +--
 3 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Mail_Domain.php b/root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Mail_Domain.php
index a8458d3..9c05d86 100644
--- a/root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Mail_Domain.php
+++ b/root/usr/share/nethesis/NethServer/Language/en/NethServer_Module_Mail_Domain.php
@@ -13,7 +13,5 @@ $L['Update domain `${0}`'] = 'Update domain `${0}`';
 $L['domain_label'] = 'Domain';
 $L['valid_relay_notprimarydomain'] = 'The local mailbox domain cannot be relayed';
 $L['OpenDkimStatus_label'] = 'Sign outbound messages with DomainKeys Identified Mail (DKIM)';
-$L['DkimKey_label'] = 'Copy and paste the following text in the record data (RDATA) section:';
-$L['DkimDNSTips_label'] = 'Add a TXT record to your public DNS service provider:';
-$L['DkimSelector_label'] = 'Set the record name to';
-
+$L['DkimSelector_label'] = '1. Add a TXT record to your public DNS service provider with key "${0}"';
+$L['DkimKey_label'] = '2. Copy and paste the following text in the record data (RDATA) section';
diff --git a/root/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php b/root/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php
index e232aa8..fb233e5 100644
--- a/root/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php
+++ b/root/usr/share/nethesis/NethServer/Module/Mail/Domain/Modify.php
@@ -171,6 +171,7 @@ class Modify extends \Nethgui\Controller\Table\Modify
             'delete' => 'Nethgui\Template\Table\Delete',
         );
         $view->setTemplate($templates[$this->getIdentifier()]);
+        $view['DkimSelector'] = 'default._domainKey';
     }
 
     /**
diff --git a/root/usr/share/nethesis/NethServer/Template/Mail/Domain/Modify.php b/root/usr/share/nethesis/NethServer/Template/Mail/Domain/Modify.php
index 93999f6..6f16b7f 100644
--- a/root/usr/share/nethesis/NethServer/Template/Mail/Domain/Modify.php
+++ b/root/usr/share/nethesis/NethServer/Template/Mail/Domain/Modify.php
@@ -44,8 +44,7 @@ echo $view->fieldsetSwitch('DisclaimerStatus', 'enabled', $view::FIELDSETSWITCH_
 
 echo $view->fieldsetSwitch('OpenDkimStatus', 'enabled', $view::FIELDSETSWITCH_EXPANDABLE | $view::FIELDSETSWITCH_CHECKBOX)
     ->setAttribute('uncheckedValue', 'disabled')
-    ->insert($view->literal($T('DkimDNSTips_label').'<br /><br />'))
-    ->insert($view->literal($T('DkimSelector_label').' : "default._domainkey"<br /><br />'))
+    ->insert($view->textLabel('DkimSelector')->setAttribute('tag', 'div')->setAttribute('template', $T('DkimSelector_label')))
     ->insert($view->textArea('DkimKey', $view::LABEL_ABOVE|$view::STATE_READONLY)->setAttribute('dimensions', '10x80'));
 
 echo $view->buttonList($view::BUTTON_SUBMIT | $view::BUTTON_HELP | $view::BUTTON_CANCEL);
-- 
2.14.3
```

DavidePrincipi reviewed (changes requested):
- Log output prefix
- Still a small fix for `sendmail` commands and similar
- Little UI tweak proposal

stephdl committed:
Added [ERROR] when the action fails

stephdl committed:
added non_smtpd_milters

stephdl committed:
UI tweak by davidep

stephdl committed:
fix PHP Warning with Nethgui\\System\\NethPlatform::getMapAdapter()

DavidePrincipi committed:
Dedicated variable for non_smtpd_milters

[skip ci]

DavidePrincipi committed:
Initialize non_smtpd_milters

DavidePrincipi committed:
Fix permissions on /var/run/opendkim directory

The permissions must be fixed in systemd-tmpfiles configuration because
/run is mounted on a tmpfs.

DavidePrincipi commented:
I opened a PR on this PR :)

stephdl committed:
Merge pull request #1 from DavidePrincipi/tmpfiles

Fix permissions on /var/run/opendkim directory
